runtime/lua/vscode-neovim/multi-cursors.lua

```diff
@@ -0,0 +1,314 @@
+local api = vim.api
+local fn = vim.fn
+
+---@class Position
+---@field line integer
+---@field character integer
+
+---@class LspRange
+---@field start Position
+---@field end Position
+
+---@class Cursor
+---@field range LspRange
+---@field extmarks number[] cursor hilight and range(selection) highlight
+
+local CREATE_CURSOR_DEFER_TIME = 10
+local START_DEFER_TIME = 40
+local NOTIFY_DEFER_TIME = 20
+
+local ns = api.nvim_create_namespace("--- vscode multiple cursors ---")
```

Comment:
nit: strange to use punctuation in a namespace name.
```suggestion
local ns = api.nvim_create_namespace("vscode.multicursor")
```